### PR TITLE
Break compile time dependency between ui-common and apps

### PR DESCRIPTION
**Pull request by belevy**
Break the compile time dependencies between ui-common and ui-driver/ui-customer to support seperate compilation

* Extract common colors used by common components instead of relying on individual apps to define common colors
* Stop using Storage module to lookup LANGUAGE_KEY in the shared preferences store
* Move FareBreakup component into ui-customer app
* Move common animation and styles types into ui-common package
* Remove unnecessary dependency on `GlobalState` type and replace with a free type variable

### CODE REVIEW

**Comment by belevy** [comment]
@srid I went ahead and broke out the trailing-ws change into its own commit as best as I could. Not sure if that is good enough.

**Comment by srid** [comment]
A separate commit is good from commit history standpoint. For the PR review standpoint, people can use "Hide whitespace" option:

image

**Review by witcher-shailesh — APPROVED** [review]

**Review by srid — COMMENTED** [review]
Since #594 is merged, this requires a rebase on top of `main`.

File: flake.nix [hunk 1]
```diff
@@ -10,6 +10,8 @@
     shared-kernel.inputs.nixpkgs.follows = "nixpkgs";
     beckn-gateway.url = "github:nammayatri/beckn-gateway";
     beckn-gateway.inputs.shared-kernel.follows = "shared-kernel";
+    easy-purescript-nix.url = "github:justinwoo/easy-purescript-nix";
+    easy-purescript-nix.flake = false;
```
Comment (srid) [inline flake.nix:14]: Group these out with a `# Frontend inputs` comment (just like the backend inputs).

**Review by srid — APPROVED** [review]
(I haven't reviewed the PureScript changes)

**Pull request merged**